sbi/inference/snpe/snpe_c.py

```diff
@@ -257,7 +258,7 @@ def _log_prob_proposal_posterior(
         theta: Tensor,
         x: Tensor,
         masks: Tensor,
-        proposal: Optional[Any],
+        proposal: DirectPosterior,
```

Comment:
Hm in principle yes. But it must implement log_prob and a `default_x`. So I don't think what a good type would be (can't use `nflows.flows.Flow` because of the default_x)

Reply:
I agree, we would need to define sth like a `Proposal` type that follows the API of our posteriors. I think we can worry about this later, the typing here makes sense.